lib/rubocop/cop/style/access_modifier_declarations.rb

```diff
@@ -130,6 +158,54 @@ def message(range)
             format(INLINE_STYLE_MESSAGE, access_modifier: access_modifier)
           end
         end
+
+        def find_corresponding_def_node(node)
+          if access_modifier_with_symbol?(node)
+            method_name = node.arguments.first.value
+            node.parent.children.find do |child|
+              child&.def_type? && child&.method?(method_name)
+            end
+          else
+            node.arguments.first
+          end
+        end
+
+        def find_argument_less_modifier_node(node)
+          node.parent.children.find do |child|
+            child&.send_type? && child&.method?(node.method_name) && child&.arguments&.empty?
+          end
```

Comment:
```suggestion
          node.parent.children.compact.find do |child|
            child.send_type? && child.method?(node.method_name) && child.arguments.empty?
          end
```

Reply:
I thought that it would be wasteful to iterate twice on `#children` for this purpose.
By the way, if we want to remove this safe navigation operators, I think it is better to use `#each_child_node` than to use `#compact`, what do you think?

```suggestion
          node.parent.each_child_node.find do |child|
            child.send_type? && child.method?(node.method_name) && child.arguments.empty?
          end
```

Reply:
I think it's a good choice.